redshift_sqlalchemy/dialect.py

```diff
@@ -145,19 +163,91 @@ class RedshiftDialect(PGDialect_psycopg2):
+        keyfunc = lambda col: abs(int(col.sortkey))
+        sortkey_cols = sorted([col for col in columns if col.sortkey],
+                              key=keyfunc)
+        interleaved = any([int(col.sortkey) < 0 for col in sortkey_cols])
+        sortkey = [col.name for col in sortkey_cols] or None
+        sortkey_type = 'interleaved' if interleaved else 'compound'
+        distkeys = [col.name for col in columns if col.distkey]
+        distkey = distkeys[0] if distkeys else None
+        return {
+            'redshift_diststyle': table_info.diststyle,
+            'redshift_distkey': distkey,
+            'redshift_sortkey': sortkey,
+            'redshift_sortkey_type': sortkey_type,
+        }
+
+    @reflection.cache
+    def get_columns(self, connection, table_name, schema=None, **kw):
+        all_columns = self._get_all_column_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        these_columns = all_columns[key]
+        domains = self._load_domains(connection)
+        columns = [self._get_column_info(col, domains) for col in these_columns]
+        return columns
+
+    @reflection.cache
     def get_pk_constraint(self, connection, table_name, schema=None, **kw):
-        """
-        Constraints in redshift are informational only. This allows reflection
-        to work.
-        """
-        return {'constrained_columns': [], 'name': ''}
+        all_constraints = self._get_all_constraint_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        constraints = all_constraints.get(key, [])
+        pk_constraints = [c for c in constraints if c.contype == 'p']
+        if not pk_constraints:
+            return {'constrained_columns': [], 'name': ''}
+        pk_constraint = pk_constraints[0]
+        # Remove the 'PRIMARY KEY(' prefix and ')' suffix
+        text = pk_constraint.condef[13:-1]
+        constrained_columns = text.split(', ')
+        return {
+            'constrained_columns': constrained_columns,
+            #'name': pk_constraint.conname,
+        }
+
+    @reflection.cache
+    def get_foreign_keys(self, connection, table_name, schema=None, **kw):
+        all_constraints = self._get_all_constraint_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        constraints = all_constraints.get(key, [])
+        fk_constraints = [c for c in constraints if c.contype == 'f']
+        fkeys = []
+        for constraint in fk_constraints:
+            print(constraint.condef)
+            m = re.match(FK_RE, constraint.condef)
+            colstring, referred_table, referred_key = m.groups()
+            referred_schema = None
+            if '.' in referred_table:
+                referred_table, referred_schema = referred_table.split('.')
```

Comment:
this will break if there is ever more than one period, but in a really obvious way, so that's probably fine.